vignettes/adrs_gcig.Rmd

```diff
@@ -0,0 +1,799 @@
+
+```{r echo=FALSE}
+list_resp <- tribble(
+  ~"Confirmed RECIST1.1 response", ~"Confirmed CA-125 response", ~"Combined response",
+  "CR", "CR", "CR",
+  "CR", "PR/SD/NE", "PR",
+  "PR/SD/NE", "CR", "PR",
+  "PR", "CR/PR/SD/NE", "PR",
+  "CR/PR/SD/NE", "PR", "PR",
+  "SD", "SD", "SD",
+  "SD", "NE", "SD",
+  "NE", "SD", "SD",
+  "PD", "any", "PD",
+  "any", "PD", "PD",
+  "NE", "NE", "NE",
+  "any", "missing", "NE",
+  "missing", "any", "NE"
+)
+
+knitr::kable(list_resp)
+```
+
+```{r}
+ovr_cbor <- covr_r11 %>%
+  derive_vars_transposed(
+    ovr_ca125,
+    by_vars = exprs(!!!get_admiral_option("subject_keys"), AVISIT, ADT, ANL01FL, ANL02FL),
+    key_var = PARAMCD,
+    value_var = AVALC
+  ) %>% 
+  select(-AVAL) %>%
+  rename(OVRR11 = AVALC) %>%
+  mutate(AVALC = case_when(
+      # CR
+      OVRR11 == "CR" & OVRCA125 == "CR" ~ "CR",
+      OVRR11 == "CR" & OVRCA125 %in% c("PR", "SD", "NE") ~ "PR",
+      OVRCA125 == "CR" & OVRR11 %in% c("PR", "SD", "NE") ~ "PR",
+      # PR
+      OVRR11 == "PR" & OVRCA125 %in% c("CR", "PR", "SD", "NE") ~ "PR",
+      OVRCA125 == "PR" & OVRR11 %in% c("CR", "PR", "SD", "NE") ~ "PR",
+      # SD, NE
+      OVRR11 %in% c("SD", "NE") & OVRCA125 %in% c("SD", "NE") &  OVRR11 == OVRCA125 ~ OVRR11,
+      OVRR11 %in% c("SD", "NE") & OVRCA125 %in% c("SD", "NE") &  OVRR11 != OVRCA125 ~ "SD",
+      # PD
+      OVRR11 == "PD" | OVRCA125 == "PD" ~ "PD",
+      # next assessment NA
+      is.na(OVRR11) | is.na(OVRCA125) ~ "NE",
+      TRUE ~ "MISSING"
+    ))
+
+ovr_cbor <- ovr_cbor %>%
+  #select(-OVRR11, -OVRCA125) %>%
+  mutate(
+    PARAMCD = "COVR11CA",
+    PARAM = "Combined Overall Response by Investigator (Confirmation of RECIST1.1 Reqired)",
+    PARAMN = 8,
+    PARCAT1 = "Investigator",
+    PARCAT1N = 1,
+    PARCAT2 = "Combined",
+    PARCAT2N = 3,
+    AVAL = aval_resp(AVALC)
+)
+  
+adrs <- bind_rows(adrs, ovr_cbor)
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, AVISIT, PARAMCD, AVALC, ADT),
+  filter = PARAMCD == "COVR11CA" & AVALC != "MISSING"
+)
+```
+
+Finally Combined Best Confirmed Overall Response parameter can be derived based on `ovr_cbor` data frame.
+
+```{r}
+ovr <- ovr_cbor
+
+adrs <- adrs %>%
+  derive_extreme_event(
+    by_vars = get_admiral_option("subject_keys"),
+    tmp_event_nr_var = event_nr,
+    order = exprs(event_nr, ADT),
+    mode = "first",
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    events = list(
+      bor_cr, bor_pr, bor_sd_gcig, bor_pd, bor_ne_gcig, no_data_missing
+    ),
+    set_values_to = exprs(
+      PARAMCD = "CBORCA11",
+      PARAM = "Combined Best Confirmed Overall Response by Investigator",
+      PARAMN = 9,
+      PARCAT1 = "Investigator",
+      PARCAT1N = 1,
+      PARCAT2 = "Combined",
+      PARCAT2N = 3,
+      AVAL = aval_resp(AVALC),
+      ANL01FL = "Y",
+      ANL02FL = "Y"
+    )
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
```

Comment:
as discussed, my understanding is different . 1 We got the same patients as responders in this logic as unconfirmed variable. but mainly as PR instead of CR for eg 1115,1118,1133 subjects.
I will check with stats and let you know when I am back.